### PR TITLE
fix(clouddriver): Fix NPE while searching for ancestor stage

### PR DESCRIPTION
**Pull request by robzienert**

### CODE REVIEW

**Comment by lwander** [comment]
should we cherry pick this into 1.6?

**Comment by ajordens** [comment]
Can you shed more light on what the pipeline looked like where this happened?

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/ApplySourceServerGroupCapacityTask.groovy [hunk 1]
```diff
@@ -158,11 +158,17 @@ class ApplySourceServerGroupCapacityTask extends AbstractServerGroupTask {
       ancestorStage.context.containsKey("sourceServerGroupCapacitySnapshot")
     }
 
-    return ancestors ? ancestors[0] : stage.execution.stages.find {
+    Stage ancestor = (ancestors != null && !ancestors.isEmpty()) ? ancestors[0] : stage.execution.stages.find {
       // find a synthetic sibling w/ 'sourceServerGroupCapacitySnapshot' in the event of there being no suitable
```
Comment (ajordens) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/providers/aws/ApplySourceServerGroupCapacityTask.groovy]: The return `ancestors ?` should have been doing the same as the revised not null and not empty check?

**Comment by robzienert** [comment]
@ajordens this isn't making pipelines fail, it appears that's the NPE is being caught and logged, but each of the executions I checked all completed successfully. It does happen super often though.

I changed the null/empty check because groovy truth isnt terribly obvious to me: it's not the cause.

**Comment by ajordens** [comment]
Did the pipeline not have the source capacity snapshot? Should it have?

If things are failing in a way that is leaving the newly created server groups with min = desired pinned, that seems like a larger issue worth tackling.

**Pull request merged**